### PR TITLE
feat(core): AWS Role Assumption support

### PR DESCRIPTION
**Pull request by n4ch04**
### Context 

Prowler 3.0 needs support to role assumption (including its settings like custom session duration and external id) 

### Description

Add support to options -A -R -I and -T of Prowler 2.X to Prowler 3.X

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/logger.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # https://docs.python.org/3/library/logging.html#logrecord-attributes
 logging.basicConfig(
     stream=sys.stdout,
-    format="%(asctime)s [File: %(filename)s] \t[Module: %(module)s]\t %(levelname)s: %(message)s",
+    format=f"%(asctime)s [File: %(filename)s:%(lineno)d] \t[Module: %(module)s]\t %(levelname)s: %(message)s",
```
Comment (jfagoagas) [inline lib/logger.py]: `f-strings` are not needed here. my bad

```suggestion
    format="%(asctime)s [File: %(filename)s:%(lineno)d] \t[Module: %(module)s]\t %(levelname)s: %(message)s",
```

File: providers/aws/aws_provider.py [hunk 2]
```diff
@@ -1,24 +1,180 @@
...
+                assumed_refreshable_credentials = RefreshableCredentials(
+                    access_key=session_info.credentials.aws_access_key_id,
+                    secret_key=session_info.credentials.aws_secret_access_key,
+                    token=session_info.credentials.aws_session_token,
+                    expiry_time=session_info.credentials.expiration,
+                    refresh_using=self.refresh,
+                    method="sts-assume-role",
+                )
+                assumed_botocore_session = get_session()
+                assumed_botocore_session._credentials = assumed_refreshable_credentials
```
Comment (jfagoagas) [inline providers/aws/aws_provider.py:73]: Document this please or include a reference to Boto3

File: providers/aws/services/iam/iam_service.py [hunk 3]
```diff
@@ -51,8 +51,9 @@ def __get_credential_report__(self):
     def __get_groups__(self):
         try:
             get_groups_paginator = self.client.get_paginator("list_groups")
-        except botocore.exceptions.ClientError as error:
-            raise error
+        except Exception as error:
+            logger.critical(f"{error.__class__.__name__} -- {error}")
+            quit()
```
Comment (jfagoagas) [inline providers/aws/services/iam/iam_service.py:68]: Why this `quit()`? Right now, in Prowler if some AWS API call fails the execution continues only showing the error message.

Also, this is not `critical`.

File: prowler.py [hunk 4]
```diff
@@ -56,14 +56,65 @@ def recover_modules_from_provider(provider):
...
+        help="AWS account id where the role passed by -R is assumed",
+    )
+    parser.add_argument(
+        "-T",
+        "--session-duration",
+        nargs="?",
+        default=3600,
+        type=int,
+        help="Assumed role session duration in seconds, by default 3600",
+    )
```
Comment (jfagoagas) [inline prowler.py:83]: Are you validating the value range for this argument somewhere?

Check this https://github.com/prowler-cloud/prowler/blob/3944ea2055b0e5bb0e1c942bd5ff64dff06a7e60/include/assume_role#L31

File: prowler.py [hunk 5]
```diff
@@ -56,14 +56,65 @@ def recover_modules_from_provider(provider):
...
-    profile = args.profile
+    if args.role or args.account:
+        if not args.account:
+            logger.error(
+                "It is needed to input an Account Id to assume the role (-A option) when a role is provided with -R"
+            )
+            quit()
+        elif not args.role:
+            logger.error(
+                "It is needed to input an role name (-R option) when an account is provided with -A"
```
Comment (jfagoagas) [inline prowler.py]:
```suggestion
                "It is needed to input an IAM Role name (-R option) when an Account Id is provided with -A"
```

File: prowler.py [hunk 6]
```diff
@@ -56,14 +56,65 @@ def recover_modules_from_provider(provider):
...
     # Parse Arguments
     args = parser.parse_args()
+
     provider = args.provider
     checks = args.checks
-    profile = args.profile
+    if args.role or args.account:
+        if not args.account:
+            logger.error(
+                "It is needed to input an Account Id to assume the role (-A option) when a role is provided with -R"
```
Comment (jfagoagas) [inline prowler.py]:
```suggestion
                "It is needed to input an Account Id to assume the role (-A option) when an IAM Role is provided with -R"
```

File: prowler.py (hunk 5 above)
Comment (jfagoagas) [inline prowler.py]: Why not something like this?
` if args.role is None or args.account is None:`

**Review by jfagoagas — CHANGES_REQUESTED** [review]
I want my tests :D

File: lib/logger.py (hunk 1 above)
Comment (n4ch04) [reply lib/logger.py]: Deleted, thanks !!

File: providers/aws/aws_provider.py (hunk 2 above)
Comment (n4ch04) [reply providers/aws/aws_provider.py:73]: Added comments

Comment (n4ch04) [reply providers/aws/aws_provider.py:73]: Please check it out

File: providers/aws/services/iam/iam_service.py (hunk 3 above)
Comment (n4ch04) [reply providers/aws/services/iam/iam_service.py:68]: Need to review it

File: prowler.py (hunk 4 above)
Comment (n4ch04) [reply prowler.py:83]: Nope, adding it

Comment (n4ch04) [reply prowler.py:83]: Added

File: prowler.py (hunk 5 above)
Comment (n4ch04) [reply prowler.py]: Done

File: providers/aws/services/iam/iam_service.py (hunk 3 above)
Comment (n4ch04) [reply providers/aws/services/iam/iam_service.py:68]: Added support via exception for ClientError in api calls

**Review by jfagoagas — APPROVED** [review]
Looks perfect!

**Review by MrCloudSec — APPROVED** [review]
Amazing job!

**Pull request merged**